### ISSUE TITLE
[Request] Add batch status and button to invalidate batch in Pro UI

### ISSUE THREAD

**Issue by jai-x** [sidekiq/sidekiq#6085]
Ruby version: `v3.0.6`
Rails version: `v7.0.4`
Sidekiq / Pro / Enterprise version(s): `v6.5.12` / `pro v5.5.8`

It would be nice to see the batch validation status and be able to invalidate the batch with a button on the Pro UI instead of having to view it via the console everytime.

Silly mock of what I'm talking about:

![Screenshot 2023-10-26 at 14 27 21](https://github.com/sidekiq/sidekiq/assets/8362491/2fded8e0-63ec-4a60-ad75-c5c29b3a7ef2)

**Comment by mperham** [comment]
My design ethos for the Web UI is that it should never be part of your application logic or workflow. The APIs are there for you to integrate into your app and its pages. If you need to invalidate a batch, you'd add a button for it in the admin area for your app.

**Comment by jai-x** [comment]
> My design ethos for the Web UI is that it should never be part of your application logic or workflow. The APIs are there for you to integrate into your app and its pages. If you need to invalidate a batch, you'd add a button for it in the admin area for your app.

With regards to this, I'm not sure I see the difference between having a button to invalidate a batch and having a button to delete/retry an enqueued jobs, but I will concede the point.

Can I still request the ablilty to just view the valid/invalid status of the batch from the UI?
I think that would be quite useful in genera,

**Comment by mperham** [comment]
Yeah, that’s reasonable, the UI should probably visually distinguish batches which have been invalidated.

**Comment by mperham** [comment]
I'm trying to figure out the requirements and use cases here. For a batch, you can `invalidate_all` or `invalidate_jids(...)`.

1. What questions is the developer trying to answer?
2. Do you care which JIDs? What do you want to know at a glance? What if there are 100,000 invalidated JIDs?
3. What do you expect to see on the `/batches` page? Should the bar change color? Should the table row be greyed out if totally invalidated?
4. What do you expect to see in the `/batches/BID` details page?

I need a lot more detail about what questions you have when you visit the Batches tab so we can figure out how the UI should answer them.

**Comment by mperham** [comment]
Overview
<img width="1020" alt="Screenshot 2023-10-27 at 11 46 14 AM" src="https://github.com/sidekiq/sidekiq/assets/2911/8e1600f2-68f0-48cc-8edf-ee728ca7e431">

**Comment by mperham** [comment]
Detail, the Invalid row is new but I don't like it.
<img width="816" alt="Screenshot 2023-10-27 at 11 51 22 AM" src="https://github.com/sidekiq/sidekiq/assets/2911/6caa01d0-1a6a-4f3c-8115-efa7fda615ce">

**Comment by jai-x** [comment]
@mperham 

The flow I am trying to support is:

1. Notice that a large long running batch is erroring due to some reason outside of code and should be cancelled
2. Go to console and call `invalidate_all` on the batch
3. Go to Sidekiq dashboard to verify the correct batch has been invalidated correctly 

To this end I would expect:

* The `/batches` page to have some indication if the batch is invalidated
  * I like the progress bar as it is
  * Greying out the row seems like a good idea
* The `/batches/<bid>` page to have some indication if the batch is invalid
   * Maybe just a new table row at the top?
   * Status : Valid/Invalid maybe?
 * Not particularly concerned about invalidation at the JID level.
   * I mainly care that the batch has been made invalid and I trust the pending JIDs will auto exit themselves since the batch jobs will have `return unless valid_within_batch?` at the top of the `perform` method
